src/hooks/async/marketplace.ts

```diff
@@ -0,0 +1,220 @@
+ * ```
+ *
+ * @param contract - an instace of a marketplace contract
+ * @param queryParams - query params to pass to the query for the sake of pagination
+ * @returns a response object that includes an array of listings
+ * @beta
+ */
+export function useMarketplaceListings(
+  contract: Marketplace | undefined,
+  filter?: MarketplaceFilter,
+) {
+  const contractAddress = contract?.getAddress();
+  return useQueryWithNetwork(
+    cacheKeys.contract.getAllListings(contractAddress, filter),
+    async () => {
+      if (contract) {
+        return contract.getAllListings(filter);
+      }
+      return [];
+    },
+    {
+      enabled: !!contract || !contractAddress,
+      keepPreviousData: true,
+    },
+  );
+}
+
+/**
+ * Use this to get a list active listings from your marketplace contract.
+ *
+ * @example
+ * ```javascript
+ * const { data: listings, isLoading, error } = useMarketplaceListings(<YourMarketplaceContractInstance>, { seller: "0x...", tokenContract: "0x...", tokenId: 1, start: 0, count: 100 });
+ * ```
+ *
+ * @param contract - an instace of a marketplace contract
+ * @param queryParams - query params to pass to the query for the sake of pagination
+ * @returns a response object that includes an array of listings
+ * @beta
+ */
+export function useMarketplaceActiveListings(
+  contract: Marketplace | undefined,
+  filter?: MarketplaceFilter,
+) {
+  const contractAddress = contract?.getAddress();
+  return useQueryWithNetwork(
+    cacheKeys.contract.getActiveListings(contractAddress, filter),
+    async () => {
+      if (contract) {
+        return contract.getActiveListings(filter);
+      }
+      return [];
+    },
+    {
+      enabled: !!contract || !contractAddress,
+      keepPreviousData: true,
+    },
+  );
+}
+
+/** **********************/
+/**     WRITE HOOKS     **/
+/** **********************/
+
+/**
+ * Use this to create a new Direct Listing on your marketplace contract.
+ *
+ * @example
+ * ```jsx
+ * const Component = () => {
+ *   const {
+ *     mutate: createDirectListing,
+ *     isLoading,
+ *     error,
+ *   } = useMarketplaceListDirect(">>YourMarketplaceContractInstance<<");
+ *
+ *   if (error) {
+ *     console.error("failed to create direct listing", error);
+ *   }
+ *
+ *   return (
+ *     <button
+ *       disabled={isLoading}
+ *       onClick={() => createDirectListing(directListingdata)}
+ *     >
+ *       Create Direct Listing!
+ *     </button>
+ *   );
+ * };
+ * ```
+ *
+ * @param contract - an instace of a contract that extends the Erc721 spec (nft collection, nft drop, custom contract that follows the Erc721 spec)
+ * @param to - an address to mint the NFT to (defaults to the connected wallet)
+ * @returns a mutation object that can be used to mint a new NFT token to the connected wallet
+ * @beta
+ */
+export function useMarketplaceListDirect(contract: Marketplace | undefined) {
```

Comment:
yeah the more I see it the more I don't like having the name of the contract in the name of the hook

this could be `useDirectListAction()` or something along those lines